src/Sulu/Component/Content/Compat/Block/BlockProperty.php

```diff
@@ -17,26 +17,15 @@
 
 /**
  * Representation of a block node in template xml.
+ *
+ * @method BlockPropertyType[] getTypes()
+ * @method addType(BlockPropertyType $type)
+ * @method BlockPropertyType getType(string $name)
+ * @method BlockPropertyType getProperties(int $index)
+ * @method BlockPropertyType initProperties(int $index, string $typeName)
```

Comment:
Why are these annotations necessary? Shouldn't they be available from the parent class?

Reply:
The problem is, that the ParentClass uses `PropertyType` and this class uses `BlockPropertyType`, which extends from `PropertyType`. The difference of those are the block settings

Reply:
Ah, I see now that the return type of those functions is different... This is basically just because we still have to support the `block` tag, and decided that something like `<property type="block">` would be too hard to maintain, right?

Reply:
`<property type="block">` is also possible, it's just another representation for `<block>` ... both result in a `BlockProperty` which will be converted to `FieldMetadata`. We decided to keep the compatibility layer nearly as it is, that's the reason I didn't combine the settings with the normal `PropertyType`

Reply:
Didn't you say at some point that for some reason `<property type="block">` is not working? Has that been fixed at some point?

Reply:
No I didn't say that, did I? That was working all the time since I asked you to review this pr

Reply:
I thought you said that once when we were discussing it before you asked me to review the PR... But in that case the issue is for me resolved.